File: samples/Cpp/TestCpp/Classes/KeypadTest/KeypadTest.cpp
#include "KeypadTest.h"

KeypadTest::KeypadTest()
{
    auto s = Director::getInstance()->getWinSize();
    auto label = LabelTTF::create("Keypad Test", "Arial", 28);
    addChild(label, 0);
    label->setPosition( Point(s.width/2, s.height-50) );

    auto listener = EventListenerKeyboard::create();
    listener->onKeyReleased = CC_CALLBACK_2(KeypadTest::onKeyReleased, this);
    
    _eventDispatcher->addEventListenerWithSceneGraphPriority(listener, this);
    
    // create a label to display the tip string
    _label = LabelTTF::create("Please press any key...", "Arial", 22);
    _label->setPosition(Point(s.width / 2, s.height / 2));
    addChild(_label, 0);
    
    _label->retain();
}

KeypadTest::~KeypadTest()
{
    _label->release();
}

void KeypadTest::onKeyReleased(EventKeyboard::KeyCode keycode, Event* event)
{
    if (keycode == EventKeyboard::KeyCode::KEY_BACKSPACE)
    {
        _label->setString("BACK clicked!");
    }
    else if (keycode == EventKeyboard::KeyCode::KEY_MENU)
    {
        _label->setString("MENU clicked!");
    }
}

void KeypadTestScene::runThisTest()
{
    auto layer = new KeypadTest();
    addChild(layer);

    Director::getInstance()->replaceScene(this);
    layer->release();
}

File: samples/lua-tests/src/KeypadTest/KeypadTest.lua
local function KeypadMainLayer()	
    local layer = cc.Layer:create() 

    local function onEnter()
        print("come in")
        local  s = cc.Director:getInstance():getWinSize()
        local  label = cc.LabelTTF:create("Keypad Test", "Arial", 28)
        layer:addChild(label, 0)
        label:setPosition(cc.p(s.width/2, s.height-50))

        local labelTip = cc.LabelTTF:create("Please press any key...", "Arial", 22)
        labelTip:setPosition(cc.p(s.width / 2, s.height / 2))
        layer:addChild(labelTip, 0)

        local function onKeyReleased(keyCode, event)
            local label = event:getCurrentTarget()
            if keyCode == cc.KeyCode.KEY_BACKSPACE then
                label:setString("BACK clicked!")
            elseif keyCode == cc.KeyCode.KEY_MENU  then
                label:setString("MENU clicked!")
            end
        end

        local listener = cc.EventListenerKeyboard:create()
        listener:registerScriptHandler(onKeyReleased, cc.Handler.EVENT_KEYBOARD_RELEASED )

        local eventDispatcher = layer:getEventDispatcher()
        eventDispatcher:addEventListenerWithSceneGraphPriority(listener, labelTip)
    end

    local function onNodeEvent(event)
        if event == "enter" then
            onEnter()
        end
    end

    layer:registerScriptHandler(onNodeEvent)

    return layer
end


function KeypadTestMain()
	cclog("KeypadTestMain")
	local scene = cc.Scene:create()
	scene:addChild(KeypadMainLayer())
	scene:addChild(CreateBackMenuItem())
	return scene
end
